src/openrct2/object/ImageTable.cpp

```diff
@@ -270,15 +270,32 @@ std::string ImageTable::FindLegacyObject(const std::string& name)
     const auto env = GetContext()->GetPlatformEnvironment();
     auto objectsPath = env->GetDirectoryPath(DIRBASE::RCT2, DIRID::OBJECT);
     auto objectPath = Path::Combine(objectsPath, name);
+    if (File::Exists(objectPath))
+    {
+        return objectPath;
+    }
+
+    std::string altName = std::string(name);
```

Comment:
Would that still create a copy of `name`? `name` needs to stay untouched.

Reply:
This will call the ```operator=(const std::string& other);``` since you did not declare it a reference type. ```std::string& altName = name``` would be a reference that will mutate ```name``` since its the same memory.

Reply:
`name` is a const, so there's no need to worry about it being modified. C++ is very strict about this, even passing it to a function or constructor that takes a non-const reference that doesn't even modify it is not allowed.